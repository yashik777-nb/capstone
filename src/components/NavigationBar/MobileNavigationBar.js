import React from "react";
import { IconButton } from "@material-ui/core";
import { Menu } from "@material-ui/icons";

class MobileNavigationBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      right: false,
    };
  }

  toggleDrawer = (anchor, open) => (event) => {
    if (
      event.type === "keydown" &&
      (event.key === "Tab" || event.key === "Shift")
    )
      return;

    this.setState({ [anchor]: open });
  };

  render() {
    return (
      <React.Fragment>
        <IconButton
          edge="end"
          aria-label="menu"
          onClick={() => this.toggleDrawer("right", true)}
        >
          <Menu />
        </IconButton>
      </React.Fragment>
    );
  }
}

export default MobileNavigationBar;
