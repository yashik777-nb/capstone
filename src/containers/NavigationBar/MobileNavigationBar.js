import React from "react";
import { NavLink } from "react-router-dom";
import {
  IconButton,
  List,
  ListItem,
  ListItemText,
  Drawer,
} from "@material-ui/core";
import { withStyles } from "@material-ui/styles";
import { Menu } from "@material-ui/icons";

const styles = (theme) => ({
  list: {
    width: 250,
  },
  linkText: {
    textDecoration: `none`,
    textTransform: `uppercase`,
    color: `black`,
  },
});

class MobileNavigationBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      right: false,
    };
  }

  toggleDrawer = (anchor, open) => {
    this.setState({ [anchor]: open });
  };

  sideDrawerList = (anchor) => (
    <div
      className={this.props.list}
      role="presentation"
      onClick={() => this.toggleDrawer(anchor, false)}
      onKeyDown={() => this.toggleDrawer(anchor, false)}
    >
      <List component="nav">
        {this.props.navLinks.map(({ title, path }) => (
          <NavLink to={path} key={title} className={this.props.linkText}>
            <ListItem button>
              <ListItemText primary={title} />
            </ListItem>
          </NavLink>
        ))}
      </List>
    </div>
  );

  render() {
    return (
      <React.Fragment>
        <IconButton
          edge="end"
          aria-label="menu"
          onClick={() => this.toggleDrawer("right", true)}
        >
          <Menu fontSize="large" style={{ color: `white` }} />
        </IconButton>
        <Drawer
          anchor="right"
          open={this.state.right}
          onClose={() => this.toggleDrawer("right", false)}
        >
          {this.sideDrawerList("right")}
        </Drawer>
      </React.Fragment>
    );
  }
}

export default withStyles(styles, { withTheme: true })(MobileNavigationBar);
