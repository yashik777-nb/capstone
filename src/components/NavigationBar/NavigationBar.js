import * as React from "react";
import { NavLink } from "react-router-dom";
import {
  AppBar,
  Toolbar,
  IconButton,
  List,
  ListItem,
  ListItemText,
  Container,
  Hidden,
} from "@material-ui/core";
import { Home } from "@material-ui/icons";
import { withStyles } from "@material-ui/styles";
import Tooltip from "@material-ui/core/Tooltip";
import MobileNavigationBar from "./MobileNavigationBar";

const navLinks = [
  { title: `About`, path: `/about`, toolTip: `About Application` },
  { title: `Issues`, path: `/issues`, toolTip: `Issues in the Applicaiton` },
  { title: `Sign In`, path: `/signin`, toolTip: `Sign In to Applciaiton` },
  { title: `Register`, path: `/register`, toolTip: `Register Yourself` },
];

const styles = (theme) => ({
  navDisplayFlex: {
    display: `flex`,
    justifyContent: `space-between`,
  },
  navbarDisplayFlex: {
    display: `flex`,
    justifyContent: `space-between`,
  },
  linkText: {
    textDecoration: `none`,
    textTransform: `uppercase`,
    color: `white`,
    "&:hover": {
      color: "black",
      fontWeight: "bold",
      textDecoration: `none`,
    },
  },
  capStoneText: {
    textDecoration: `none`,
    color: `white`,
    "&:hover": {
      color: "black",
      textDecoration: `none`,
    },
  },
  homeButton: {
    "&:hover": {
      color: "black",
      anchor: "open",
    },
  },
});

class Header extends React.Component {
  render() {
    const { classes } = this.props;
    return (
      <AppBar position="static">
        <Toolbar>
          <Container maxWidth="md" className={classes.navbarDisplayFlex}>
            <Tooltip disableFocusListener title="Home">
              <IconButton
                edge="start"
                color="inherit"
                aria-label="home"
                className={classes.homeButton}
              >
                <Home fontSize="large" />
                <NavLink className={classes.capStoneText} to="/">
                  CapStone
                </NavLink>
              </IconButton>
            </Tooltip>
            <Hidden smDown>
              <List
                component="nav"
                aria-labelledby="main navigation"
                className={classes.navDisplayFlex}
              >
                {navLinks.map(({ title, path, toolTip }) => (
                  <NavLink to={path} key={title} className={classes.linkText}>
                    <Tooltip disableFocusListener title={toolTip}>
                      <ListItem button>
                        <ListItemText primary={title} />
                      </ListItem>
                    </Tooltip>
                  </NavLink>
                ))}
              </List>
            </Hidden>
            <Hidden mdUp>
              <MobileNavigationBar navLinks={navLinks} />
            </Hidden>
          </Container>
        </Toolbar>
      </AppBar>
    );
  }
}

export default withStyles(styles, { withTheme: true })(Header);
