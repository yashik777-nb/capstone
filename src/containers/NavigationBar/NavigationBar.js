import * as React from "react";
import { NavLink } from "react-router-dom";
import { connect } from "react-redux";
import {
  AppBar,
  Toolbar,
  IconButton,
  List,
  ListItem,
  ListItemText,
  Container,
  Hidden,
} from "@material-ui/core";
import { Home } from "@material-ui/icons";
import { withStyles } from "@material-ui/styles";
import Tooltip from "@material-ui/core/Tooltip";
import MobileNavigationBar from "./MobileNavigationBar";
import Search from "./Search";
import * as actionCreators from "../../store/actions/actionsIndex";

const navLinks = [
  { title: `About`, path: `/about`, toolTip: `About Application` },
  { title: `Issues`, path: `/issues`, toolTip: `Issues in the Applicaiton` },
  { title: `SignIn`, path: `/signin`, toolTip: `Sign In to Applciaiton` },
  { title: `Register`, path: `/register`, toolTip: `Register Yourself` },
];

const styles = (theme) => ({
  navDisplayFlex: {
    display: `flex`,
    justifyContent: `space-around`,
  },
  navbarDisplayFlex: {
    display: `flex`,
    justifyContent: `space-around`,
  },
  linkText: {
    textDecoration: `none`,
    textTransform: `uppercase`,
    margin: 0,
    padding: 0,
    color: `white`,
    "&:hover": {
      color: "black",
      fontWeight: "bold",
      textDecoration: `none`,
    },
  },
  capStoneText: {
    textDecoration: `none`,
    color: `white`,
    "&:hover": {
      color: "black",
      textDecoration: `none`,
    },
  },
  homeButton: {
    "&:hover": {
      color: "black",
      anchor: "open",
    },
  },
});

export class NavigationBar extends React.Component {
  render() {
    const { classes } = this.props;
    return (
      <AppBar position="static">
        <Toolbar>
          <Container maxWidth="lg" className={classes.navbarDisplayFlex}>
            <Tooltip disableFocusListener title="Home">
              <IconButton
                edge="start"
                color="inherit"
                aria-label="home"
                className={classes.homeButton}
              >
                <Home fontSize="large" />
                <NavLink className={classes.capStoneText} to="/">
                  CapStone
                </NavLink>
              </IconButton>
            </Tooltip>
            <Search filterFunction={this.props.onSearchIssue} />
            <Hidden smDown>
              <List
                component="nav"
                aria-labelledby="main navigation"
                className={classes.navDisplayFlex}
              >
                {navLinks.map(({ title, path, toolTip }) => (
                  <NavLink to={path} key={title} className={classes.linkText}>
                    <Tooltip disableFocusListener title={toolTip}>
                      <ListItem button>
                        <ListItemText primary={title} />
                      </ListItem>
                    </Tooltip>
                  </NavLink>
                ))}
                {this.props.authenticated ? (
                  <NavLink
                    to="/addissue"
                    key="Add Issue"
                    className={classes.linkText}
                  >
                    <Tooltip disableFocusListener title="Add An Issue">
                      <ListItem button>
                        <ListItemText primary="Add Issue" />
                      </ListItem>
                    </Tooltip>
                  </NavLink>
                ) : null}
                {this.props.authenticated ? (
                  <NavLink
                    to="/"
                    key="Logout"
                    className={classes.linkText}
                    onClick={(e) => this.props.logoutUser()}
                  >
                    <Tooltip
                      disableFocusListener
                      title="Logout from Application"
                    >
                      <ListItem button>
                        <ListItemText primary="Logout" />
                      </ListItem>
                    </Tooltip>
                  </NavLink>
                ) : null}
              </List>
            </Hidden>
            <Hidden mdUp>
              <MobileNavigationBar
                authenticated={this.props.authenticated}
                navLinks={navLinks}
                logout={this.props.logoutUser}
              />
            </Hidden>
          </Container>
        </Toolbar>
      </AppBar>
    );
  }
}

// mapStateToProps
const mapStateToProps = (state) => {
  return {
    authenticated: state.users.authenticated,
  };
};

const mapDispathToProps = (dispatch) => {
  return {
    onSearchIssue: (value) => dispatch(actionCreators.filterIssues(value)),
    logoutUser: () => dispatch(actionCreators.logoutUser()),
  };
};

export default withStyles(styles, { withTheme: true })(
  connect(mapStateToProps, mapDispathToProps)(NavigationBar)
);
