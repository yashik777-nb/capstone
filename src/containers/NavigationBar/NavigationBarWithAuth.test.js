import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createMount } from "@material-ui/core/test-utils";
import { createMuiTheme, MuiThemeProvider } from "@material-ui/core/styles";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";

import NavigationBar from "./NavigationBar";

configure({ adapter: new Adapter() });

const styles = {
  navDisplayFlex: "Connect(NavigationBar)-navbarDisplayFlex-58",
  navbarDisplayFlex: "Connect(NavigationBar)-navDisplayFlex-57",
  linkText: "Connect(NavigationBar)-linkText-59",
  capStoneText: "Connect(NavigationBar)-capStoneText-60",
  homeButton: "Connect(NavigationBar)-homeButton-61",
};

const store = {
  users: {
    authenticated: true,
  },
};

const props = {
  classes: styles,
  onSearchIssue: () => {},
  logoutUser: () => {},
  authenticated: true,
};

const mockStore = configureStore();

const theme = createMuiTheme({
  props: { MuiWithWidth: { initialWidth: "lg" } },
});

describe("Navigation Bar Tests", () => {
  let wrapper;

  beforeEach(() => {
    const render = createMount();
    wrapper = render(
      <Provider store={mockStore(store)}>
        <BrowserRouter>
          <MuiThemeProvider theme={theme}>
            <NavigationBar {...props} />
          </MuiThemeProvider>
        </BrowserRouter>
      </Provider>
    );
  });

  it("Should render seven NavLinks if authenticated", () => {
    expect(wrapper.find("NavLink").length).toEqual(7);
  });

  it("Navlink path check", () => {
    expect(wrapper.find("NavLink").get(0).props.to).toEqual("/");
  });

  it("Navlink path check for Sign In", () => {
    expect(wrapper.find("NavLink").get(5).props.to).toEqual("/addissue");
  });

  test("Should Render navigation Bar", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
