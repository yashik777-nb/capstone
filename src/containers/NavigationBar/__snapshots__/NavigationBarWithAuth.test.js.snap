// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar Tests Should Render navigation Bar 1`] = `ReactWrapper {}`;
